joint_trajectory_controller/include/joint_trajectory_controller/init_joint_trajectory.h

```diff
@@ -295,93 +313,156 @@ Trajectory initJointTrajectory(const trajectory_msgs::JointTrajectory&       msg
   // Initialize result trajectory: combination of:
   // - Useful segments of currently followed trajectory
   // - Useful segments of new trajectory (contained in ROS message)
-  Trajectory result_traj; // Currently empty
-
-  // Initialize offsets due to wrapping joints to zero
-  std::vector<Scalar> position_offset(n_joints, 0.0);
+  Trajectory result_traj;
 
-  // Bridge current trajectory to new one
   if (has_current_trajectory)
   {
-    const Trajectory& curr_traj = *(options.current_trajectory);
-
-    // Get the last time and state that will be executed from the current trajectory
-    const typename Segment::Time last_curr_time = std::max(o_msg_start_time.toSec(), o_time.toSec()); // Important!
-    typename Segment::State last_curr_state;
-    sample(curr_traj, last_curr_time, last_curr_state);
+    result_traj = *(options.current_trajectory);
 
-    // Get the first time and state that will be executed from the new trajectory
-    const typename Segment::Time first_new_time = o_msg_start_time.toSec() + (it->time_from_start).toSec();
-    typename Segment::State first_new_state(*it, permutation_vector); // Here offsets are not yet applied
-
-    // Compute offsets due to wrapping joints
-    if (has_angle_wraparound)
+    //Iterate to all segments to set the new goal handler
+    for (unsigned int joint_id=0; joint_id < joint_names.size();joint_id++)
     {
-      position_offset = wraparoundOffset(last_curr_state.position,
-                                         first_new_state.position,
-                                         *(options.angle_wraparound));
-      if (position_offset.empty())
+      for (unsigned int segment_id=0; segment_id < result_traj[joint_id].size(); segment_id++)
       {
-        ROS_ERROR("Cannot create trajectory from message. "
-                  "Vector specifying whether joints wrap around has an invalid size.");
-        return Trajectory();
+        (result_traj[joint_id])[segment_id].setGoalHandle(options.rt_goal_handle);
```

Comment:
The new goal handle should not be applied to the trajectory currently being executed. Only to the segments of the new one ,and to the bridge segment that continuously transitions between the current and new trajectories, which is done below.